SketchUp/Geom/PolygonMesh.rb

```diff
@@ -51,20 +51,71 @@ class Geom::PolygonMesh
   def add_point(point)
   end
 
-  # The {#add_polygon} method is used for adding a polygon to a polygon mesh.
-  #
-  # @example
-  #   mesh = Geom::PolygonMesh.new
-  #   point1 = Geom::Point3d.new(0, 1, 2)
-  #   point2 = Geom::Point3d.new(1, 0, 2)
-  #   point3 = Geom::Point3d.new(2, 0, 1)
-  #   index = mesh.add_polygon(point1, point2, point3)
-  #
-  # @param [Array<Integer>] points
-  #   A list of points that represent the
-  #   polygon's vertices.
-  #
-  # @return [Integer] the index of the polygon in the mesh
+  # The {#add_polygon} method is used for adding a {#polygons polygon} to a
+  # {PolygonMesh}. All variations of this method require at least 3 elements
+  # to define a polygon, although more may be given.
+  #
+  #
+  # @overload add_polygon(index, index, index, ...)
+  #   @example
+  #     mesh = Geom::PolygonMesh.new
+  #     # add points to mesh...
+  #     mesh.add_point(Geom::Point3d.new(0, 0, 0))
+  #     mesh.add_point(Geom::Point3d.new(1, 0, 0))
+  #     mesh.add_point(Geom::Point3d.new(1, 1, 0))
+  #     polygon_index = mesh.add_polygon(1, 2, 3)
+  #   @param [Integer] index 
+  #     An index of a vertex in the mesh. Remember, mesh indices start at 1.
+  #   @param [Integer] ... 
+  #     More (optional) indices.
+  #
+  # @overload add_polygon(index_array)
+  #   @example
+  #     mesh = Geom::PolygonMesh.new
+  #     # add points to mesh...
+  #     mesh.add_point(Geom::Point3d.new(0, 0, 0))
+  #     mesh.add_point(Geom::Point3d.new(1, 0, 0))
+  #     mesh.add_point(Geom::Point3d.new(1, 1, 0))
+  #     polygon_index = mesh.add_polygon([1, 2, 3])
+  #   @param [Array<Integer>] index_array 
+  #     An {Array} of point indices.
+  #
+  # @overload add_polygon(point3d, point3d, point3d, ...)
+  #   @example
+  #     mesh = Geom::PolygonMesh.new
+  #     point1 = Geom::Point3d.new(0, 1, 2)
+  #     point2 = Geom::Point3d.new(1, 0, 2)
+  #     point3 = Geom::Point3d.new(2, 0, 1)
+  #     polygon_index = mesh.add_polygon(point1, point2, point3)
+  #   @param [Geom::Point3d] point3d 
+  #     Note when passing a flat list of Point3ds, arrays can not be
+  #     substituted and will generate an error:
+  #     +#<ArgumentError: point index 0 out of range>+.
```

Comment:
`+#<ArgumentError: point index 0 out of range>+` Doesn't seem to format right in my test. However, I think this can be added to a `@raise` tag for this overload that describe this error.